### ISSUE TITLE
No warning for missing cilk return

### ISSUE THREAD

**Issue by krame505** [melt-umn/ableC-cilk#8]
I don't have time to look at this in more detail, but I added a simple recursive fold test program to the examples, which crashes with an error about a bad free or Cilk internal error, after printing the result value - seemingly on exit from main.  Could this have something to do with the use of `malloc` in this example?

**Comment by TravisCarlson** [comment]
Try adding `cilk return 0;` at the end of `main`?

**Comment by krame505** [comment]
Ah, I didn't realize that was required.  Presumably we should produce some sort of a warning in that case?

**Comment by ericvanwyk** [comment]
Yes.  Don't close this until we fix the Cilk extension to do so.
